modules/phase_field/src/userobjects/EBSDReader.C

```diff
@@ -244,123 +246,85 @@ EBSDReader::readFile()
     a._p += j._p;
     a._n++;
   }
-  // creating a vector of maps to store the quaternion count for each bin index
-  std::vector<
-      std::map<std::tuple<unsigned int, unsigned int, unsigned int, unsigned int>, unsigned int>>
-      feature_weights(_grain_num);
-  // creating a vector of vectors to store the bin index corresponding to a quaternion
-  std::vector<std::vector<std::tuple<unsigned int, unsigned int, unsigned int, unsigned int>>>
-      quat_to_bin(_grain_num);
-
-  for (unsigned int i = 0; i < _grain_num; i++)
+
+  for (const auto i : make_range(_grain_num))
   {
     EBSDAvgData & a = _avg_data[i];
     EulerAngles & b = _avg_angles[i];
 
     if (a._n == 0)
       continue;
 
-    Real w_index, x_index, y_index, z_index;
-    Real w, x, y, z;
-    std::vector<std::tuple<double, Eigen::VectorXd>> eigen_vectors_and_values;
+    // creating a map to store the quaternion count for each bin index
+    std::map<std::tuple<int, int, int, int>, unsigned int> feature_weights;
 
-    // creating an empty map
-    feature_weights.push_back(
-        std::map<std::tuple<unsigned int, unsigned int, unsigned int, unsigned int>,
-                 unsigned int>());
-
-    // looping through quaternions of each grain
-    for (unsigned int k = 0; k < quat[i].size(); k++)
+    // looping through all quaternions of the current grain
```

Comment:
~~How does a single grain have more than one quaternion vector associated? (for my own edification)~~

Reply:
If this is just looping over all vector components of the quaternion: Does it make sense to hard code a "4" somehow instead of interrogating its `size`?

Reply:
Not vector components, but all quaternions on sample point in the EBSD file which belong to a specific grain.